From d6e56f8ceffc6a7ba238b58f078d3329e9d260fc Mon Sep 17 00:00:00 2001
From: Piyush Gupta <gpiyush@vmware.com>
Date: Thu, 9 Dec 2021 10:24:01 +0000
Subject: [PATCH] selectdisk.py: Display appropriate error if no block device
 found.

Added a new window to display error if there is no devices attached.

Change-Id: If7fabceec00dd124c6c7c3e2cbc38ef1add99cb7
---
 photon_installer/selectdisk.py | 17 ++++++++++++++---
 1 file changed, 14 insertions(+), 3 deletions(-)

diff --git a/photon_installer/selectdisk.py b/photon_installer/selectdisk.py
index fbe66d7..f6870e1 100755
--- a/photon_installer/selectdisk.py
+++ b/photon_installer/selectdisk.py
@@ -6,6 +6,7 @@
 #
 #    Author: Mahmoud Bassiouny <mbassiouny@vmware.com>
 
+import sys
 from device import Device
 from window import Window
 from actionresult import ActionResult
@@ -38,13 +39,23 @@ class SelectDisk(object):
         self.devices = None
 
     def display(self):
-        self.window.addstr(0, 0, 'Please select a disk and a method how to partition it:\n' +
-                           'Auto - single partition for /, no swap partition.\n' +
-                           'Custom - for customized partitioning')
 
         self.disk_menu_items = []
 
         self.devices = Device.refresh_devices()
+
+        if len(self.devices) == 0:
+            err_win = Window(self.win_height, self.win_width, self.maxy, self.maxx,
+                             'Select a disk', False, position=2, tab_enabled=False)
+            err_win.addstr(0, 0, 'No block devices found to select\n' +
+                           'Press any key to get to bash.')
+            err_win.show_window()
+            err_win.content_window().getch()
+            sys.exit(1)
+
+        self.window.addstr(0, 0, 'Please select a disk and a method how to partition it:\n' +
+                           'Auto - single partition for /, no swap partition.\n' +
+                           'Custom - for customized partitioning')
         # Fill in the menu items
         for index, device in enumerate(self.devices):
             #if index > 0:
-- 
2.23.3

